### ISSUE TITLE
IGMP/Multicast traffic not working on 24.7 and Intel X710 NIC

### ISSUE THREAD

**Issue by Hipoglos** [opnsense/src#212]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

IGMP/Multicast traffic not working on 24.7 and Intel x710 NIC

*Tip*: to validate your setup was working with the previous version, use opnsense-revert (https://docs.opnsense.org/manual/opnsense_tools.html#opnsense-revert)

**To Reproduce**

1. Install and configure IGMP Proxy, using an Intel x710 NIC
2. Configure L2 accordingly
3. Monitor IPTV Multicast traffic 

**Expected behavior**

Traffic should be tagged accordingly

**Describe alternatives you considered**

Same configuration, but different NIC will work fine, so it seams a IGMP Proxy compability issue with the Intel NIC.

I've tested also 24.7_10 kernel, but the issue remains.

**Screenshots**

N/A

**Relevant log files**

<5>ixl0: Link is up, 1 Gbps Full Duplex, Requested FEC: None, Negotiated FEC: None, Autoneg: True, Flow Control: None
<6>ixl0: link state changed to UP

<5>ixl1: Link is up, 1 Gbps Full Duplex, Requested FEC: None, Negotiated FEC: None, Autoneg: True, Flow Control: None
<6>ixl1: link state changed to UP

root@opnsense:~ # sysctl dev.ixl.0.fw_version ; sysctl dev.ixl.0.iflib.driver_version
dev.ixl.0.fw_version: fw 9.150.77492 api 1.15 nvm 9.50 etid 8000f12c oem 0.0.0
dev.ixl.0.iflib.driver_version: 2.3.3-k

* WAN 
root@opnsense:~ # ifconfig -vv ixl0
ixl0: flags=1008a43<UP,BROADCAST,RUNNING,ALLMULTI,SIMPLEX,MULTICAST,LOWER_UP> metric 0 mtu 1500
        description: ONT (opt11)
        options=4800028<VLAN_MTU,JUMBO_MTU,HWSTATS,MEXTPG>
        ether xx:xx:xx:xx:xx:xx
        inet6 xx::xx:xx:xx:xx%ixl0 prefixlen 64 scopeid 0x1
        groups: Interfaces
        media: Ethernet autoselect (1000baseLX <full-duplex>)
        status: active
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        drivername: ixl0
        plugged: SFP/SFP+/SFP28 1000BASE-LX (SC)
        vendor: FS PN: SFP-GE-BX SN: C2401543867 DATE: 2024-01-30
        module temperature: 46.89 C voltage: 3.27 Volts
        lane 1: RX power: 0.21 mW (-6.82 dBm) TX bias: 21.84 mA

* LAN 
root@opnsense:~ # ifconfig -vv ixl1
ixl1: flags=1008a43<UP,BROADCAST,RUNNING,ALLMULTI,SIMPLEX,MULTICAST,LOWER_UP> metric 0 mtu 1500
        description: LAN (lan)
        options=4900028<VLAN_MTU,JUMBO_MTU,NETMAP,HWSTATS,MEXTPG>
        ether XX:XX:XX:XX:XX:XX
        inet 10.0.0.1 netmask 0xffffff00 broadcast 10.0.0.255
        inet6 xxxx::xxxx:xxxx:xxxx:xxxx%ixl1 prefixlen 64 scopeid 0x2
        inet6 xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxxprefixlen 64
        groups: Interfaces
        media: Ethernet autoselect (1000baseT <full-duplex>)
        status: active
        nd6 options=21<PERFORMNUD,AUTO_LINKLOCAL>
        drivername: ixl1
        plugged: SFP/SFP+/SFP28 1000BASE-T (RJ45)
        vendor: Ubiquiti Inc. PN: CM-RJ45-1G SN: XXXXXXXXXXX DATE: 2024-01-03

ixl0@pci0:2:0:0:        class=0x020000 rev=0x02 hdr=0x00 vendor=0x8086 device=0x1572 subvendor=0x8086 subdevice=0x0000
    vendor     = 'Intel Corporation'
    device     = 'Ethernet Controller X710 for 10GbE SFP+'
    class      = network
    subclass   = ethernet

ixl1@pci0:2:0:1:        class=0x020000 rev=0x02 hdr=0x00 vendor=0x8086 device=0x1572 subvendor=0x8086 subdevice=0x0000
    vendor     = 'Intel Corporation'
    device     = 'Ethernet Controller X710 for 10GbE SFP+'
    class      = network
    subclass   = ethernet

**Additional context**

- I'm using a PPPoE FTTH WAN connection, and specific VLAN over WAN for IPTV traffic.

- When restarting IGMP Proxy service:

*OPNSense Log*

2024-08-05T12:02:27	Notice	kernel	<6>ixl1: Disabled multicast promiscuous mode	
2024-08-05T12:02:27	Notice	kernel	<6>ixl1: Disabled multicast promiscuous mode	
2024-08-05T12:02:27	Notice	kernel	<6>ixl1: Disabled multicast promiscuous mode
2024-08-05T12:02:27	Warning	igmpproxy	select() failure; Errno(4): Interrupted system call

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 24.7_9 (amd64).
Intel® Core™ i5-12600H
Network Intel® x710

**Comment by fichtner** [comment]
Forum link https://forum.opnsense.org/index.php?topic=41923.0

**Comment by fichtner** [comment]
89e7335942 looks suspicious

**Comment by fichtner** [comment]
So try this kernel then:

    # opnsense-update zkr 24.7.1-ixl


Cheers,
Franco

**Comment by Hipoglos** [comment]
@fichtner Franco, just tested and it works!  Thanks a lot! It’s late now, but i will continue testing it tomorrow and let you know. You can close it as resolved 🫶🏻

**Comment by fichtner** [comment]
@Hipoglos thanks! unfortunately this is where the real work begins :)

**Comment by fichtner** [comment]
Made a note here: https://reviews.freebsd.org/D40860#1054462

Intel is pretty good at following up so let's wait and see :)

Note the kernel is based on 24.7.1 but when you install 24.7.1 it will want to go back to the bad kernel so either lock it from the packages tab or reapply the kernel afterwards again (though keeping it unlocked will present the same issue for each update).


Cheers,
Franco

**Comment by Hipoglos** [comment]
If I can help with further testing or anything you may need, just let me know. Always happy to help.

**Comment by fichtner** [comment]
@Hipoglos thanks, will keep this open and let you know

**Comment by maximushugus** [comment]
There is the same bug in pfSense (CE 2.7.1, CE 2.7.2, and probably pfSense+).
Here is the bug in [netgate forum](https://forum.netgate.com/topic/188736/multicast-inconsistant) and in [pfSense bug tracker](https://redmine.pfsense.org/issues/15679).

**Comment by fichtner** [comment]
Sounds probable. FreeBSD current and 14.0 and 14.1 all carry the buggy commit.

**Comment by maximushugus** [comment]
@fichtner I can see that someone from Intel responded on your comment [here](https://reviews.freebsd.org/D40860#1054462)
I do not have an account on reviews.freebsd.org but I can respond here to try to help.

To describe my config : pfSense (so not opnsense but same freebsd and same bug) with an intel X710 NIC. I have a LAGG of ixl0 and ixl1.
Before running those commands I lauched IGMP Proxy on my pfSense and joined a multicast group (233.136.0.202) on a PC on my LAN.

Here are the results : 

```shell
[2.7.2-RELEASE][root@pfSense.home.arpa]/root: sysctl dev.ixl.0.debug.filter_list
dev.ixl.0.debug.filter_list: 
PF Filters:
01:80:c2:00:00:02, vlan   -1, flags 0x0002
00:25:90:XX:XX:XX, vlan    0, flags 000000
00:25:90:XX:XX:XX, vlan    3, flags 0x0001
00:25:90:XX:XX:XX, vlan   50, flags 0x0001
00:25:90:XX:XX:XX, vlan   99, flags 0x0001
00:25:90:XX:XX:XX, vlan    1, flags 0x0001
00:25:90:XX:XX:XX, vlan    2, flags 0x0001
33:33:ff:XX:XX:XX, vlan   -1, flags 0x0002
33:33:00:00:00:01, vlan   -1, flags 0x0002
33:33:ff:18:61:20, vlan   -1, flags 0x0002
33:33:18:61:20:ce, vlan   -1, flags 0x0002
01:00:5e:00:00:01, vlan   -1, flags 0x0002
33:33:ff:00:00:01, vlan   -1, flags 0x0002
33:33:00:00:00:02, vlan   -1, flags 0x0002
33:33:00:00:00:fb, vlan   -1, flags 0x0002
01:00:5e:00:00:fb, vlan   -1, flags 0x0002
01:00:5e:00:00:02, vlan   -1, flags 0x0002
01:00:5e:00:00:16, vlan   -1, flags 0x0002
``` 

```shell
[2.7.2-RELEASE][root@pfSense.home.arpa]/root: sysctl dev.ixl.1.debug.filter_list
dev.ixl.1.debug.filter_list: 
PF Filters:
01:80:c2:00:00:02, vlan   -1, flags 0x0002
00:25:90:XX:XX:XX, vlan    0, flags 000000
00:25:90:XX:XX:XX, vlan    3, flags 0x0001
00:25:90:XX:XX:XX, vlan   50, flags 0x0001
00:25:90:XX:XX:XX, vlan   99, flags 0x0001
00:25:90:XX:XX:XX, vlan    1, flags 0x0001
00:25:90:XX:XX:XX, vlan    2, flags 0x0001
33:33:ff:XX:XX:XX, vlan   -1, flags 0x0002
33:33:00:00:00:01, vlan   -1, flags 0x0002
33:33:ff:18:61:20, vlan   -1, flags 0x0002
33:33:18:61:20:ce, vlan   -1, flags 0x0002
01:00:5e:00:00:01, vlan   -1, flags 0x0002
33:33:ff:00:00:01, vlan   -1, flags 0x0002
33:33:00:00:00:02, vlan   -1, flags 0x0002
33:33:00:00:00:fb, vlan   -1, flags 0x0002
01:00:5e:00:00:fb, vlan   -1, flags 0x0002
01:00:5e:00:00:02, vlan   -1, flags 0x0002
01:00:5e:00:00:16, vlan   -1, flags 0x0002
```

and for ifmcstat (I do not show interfaces that are not related with this bug) : 

> 
```shell
[2.7.2-RELEASE][root@pfSense.home.arpa]/root: ifmcstat
ixl0:
ixl1:
lo0:
        inet 127.0.0.1
        igmpv3 rv 2 qi 125 qri 10 uri 3
                group 224.0.0.1 mode exclude
        inet6 fe80::1%lo0 scopeid 0x5
        mldv2 flags=2<USEALLOW> rv 2 qi 125 qri 10 uri 3
                group ff01::1%lo0 scopeid 0x5 mode exclude
                group ff02::2:1861:20ce%lo0 scopeid 0x5 mode exclude
                group ff02::2:ff18:6120%lo0 scopeid 0x5 mode exclude
                group ff02::1%lo0 scopeid 0x5 mode exclude
                group ff02::1:ff00:1%lo0 scopeid 0x5 mode exclude
lagg0:
        inet6 fe80::225:90ff:fe9c:5644%lagg0 scopeid 0xa
        mldv2 flags=2<USEALLOW> rv 2 qi 125 qri 10 uri 3
                group ff01::1%lagg0 scopeid 0xa mode exclude
                        mcast-macaddr 33:33:00:00:00:01
                group ff02::2:1861:20ce%lagg0 scopeid 0xa mode exclude
                        mcast-macaddr 33:33:18:61:20:ce
                group ff02::2:ff18:6120%lagg0 scopeid 0xa mode exclude
                        mcast-macaddr 33:33:ff:18:61:20
                group ff02::1%lagg0 scopeid 0xa mode exclude
                        mcast-macaddr 33:33:00:00:00:01
                group ff02::1:ff9c:5644%lagg0 scopeid 0xa mode exclude
                        mcast-macaddr 33:33:ff:XX:XX:XX
lagg0.3:
        inet 192.168.3.1
        igmpv2
                group 224.0.0.22 mode exclude
                        mcast-macaddr 01:00:5e:00:00:16
                group 224.0.0.2 mode exclude
                        mcast-macaddr 01:00:5e:00:00:02
                group 224.0.0.251 mode exclude
                        mcast-macaddr 01:00:5e:00:00:fb
        inet6 fe80::225:90ff:fe9c:5644%lagg0.3 scopeid 0xb
        mldv2 flags=2<USEALLOW> rv 2 qi 125 qri 10 uri 3
                group ff02::fb%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:00:00:00:fb
                group ff02::2%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:00:00:00:02
        inet 192.168.3.1
        igmpv2
                group 224.0.0.1 mode exclude
                        mcast-macaddr 01:00:5e:00:00:01
        inet6 fe80::225:90ff:fe9c:5644%lagg0.3 scopeid 0xb
        mldv2 flags=2<USEALLOW> rv 2 qi 125 qri 10 uri 3
                group ff02::1:ff00:1%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:ff:00:00:01
                group ff01::1%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:00:00:00:01
                group ff02::2:1861:20ce%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:18:61:20:ce
                group ff02::2:ff18:6120%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:ff:18:61:20
                group ff02::1%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:00:00:00:01
                group ff02::1:ff9c:5644%lagg0.3 scopeid 0xb mode exclude
                        mcast-macaddr 33:33:ff:XX:XX:XX
```

**Comment by fichtner** [comment]
Reported upstream https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=281125

**Comment by fichtner** [comment]
I actually added the patch to stable/24.7 now for the next 24.7.x iteration given the fact that the author from Intel's second response was "I don't work here anymore". This could take a lot longer than expected. ;)


Cheers,
Franco

**Comment by fichtner** [comment]
Closing because fixed with the revert in 24.7.4 for the foreseeable future.
